controller/src/main/java/io/pravega/controller/store/task/AbstractTaskMetadataStore.java

```diff
@@ -80,4 +88,51 @@ abstract Void transferLock(final Resource resource,
                               final String oldThreadId);
 
     abstract Void removeLock(final Resource resource, final String owner, final String tag);
+
+    @Override
+    public CompletableFuture<Void> putChild(final String parent, final TaggedResource child) {
```

Comment:
I'm a bit confused by the concept of parent/child in these calls. I initially thought that this was related to ZK, but then I saw that this was originally in the `InMemoryTaskMetadataStore`. What's this referring to?